tests/integration/release/test_release.py

```diff
@@ -112,92 +131,332 @@ def test_full_release_noising(
 
-    for dataset in datasets:
-        dataset._clean_input_data()
-        dataset._reformat_dates_for_noising()
+            elif isinstance(noise_type, ColumnNoiseType):
+                for column in dataset_schema.columns:
+                    if config.has_noise_type(
+                        dataset_schema.name, noise_type.name, column.name
+                    ):
+                        # don't noise ssa_event_type because it's used as an identifier column
+                        # along with simulant id
+                        # TODO: Noise ssa_event_type when record IDs are implemented (MIC-4039)
+                        if column == COLUMNS.ssa_event_type:
+                            continue
+                        data = data.map_partitions(
+                            apply_column_noise_type,
+                            column=column,
+                            noise_type=noise_type,
+                            config=config,
+                            meta=noised_metadata,
+                        ).persist()
+
+                        missingness_correct = data.map_partitions(
+                            lambda data_: pd.Series(
+                                (
+                                    get_missingness_data(data_).values
+                                    == Dataset.is_missing(get_noised_data(data_)).values
+                                ).all()
+                            )
+                        )
+                        with check:
+                            assert missingness_correct.all().compute()
+
+                        counts = data.map_partitions(
+                            lambda data_: get_column_noising_counts(
+                                data_,
+                                config,
+                                dataset_schema.name,
+                                noise_type.name,
+                                column.name,
+                            ),
+                            meta=pd.DataFrame(
+                                columns=[
+                                    "numerator",
+                                    "denominator",
+                                    "expected_numerator",
+                                    "missing_data_not_missing",
+                                ],
+                                dtype=int,
+                            ),
+                        )
+                        total_counts = counts.sum().compute()
+                        with check:
+                            assert total_counts["missing_data_not_missing"] == 0
+                        numerator = total_counts["numerator"]
+                        denominator = total_counts["denominator"]
+                        # we can get no eligible rows for unit number in acs RI data
+                        # because it's a sparse column and acs is small
+                        if column == COLUMNS.unit_number and denominator == 0:
```

Comment:
Is this just because unit number has such little noise applied due to being such a small str?

Reply:
Most rows don't have a unit number